include/akushon/action/model/action.hpp

```diff
@@ -57,11 +59,19 @@ class Action
 
   void reset();
 
+  void enable_spline(bool enable);
+  bool is_using_spline() const;
+  void generate_splines();
+
+  std::map<uint8_t, keisan::SmoothSpline *> joint_splines;
```

Comment:
Use [`std::unordered_map`](https://en.cppreference.com/w/cpp/container/unordered_map) instead, it's faster:
```suggestion
  std::unordered_map<uint8_t, keisan::SmoothSpline *> joint_splines;
```